openedx/core/djangoapps/waffle_utils/tests/test_views.py

```diff
@@ -31,19 +35,97 @@ def test_failure_for_non_staff(self):
 
     @override_waffle_flag(TEST_WAFFLE_FLAG, True)
     def test_response_with_waffle_flag(self):
-        response = self._get_toggle_state_response(is_staff=True)
+        response = self._get_toggle_state_response()
         self.assertIn('waffle_flags', response.data)
         self.assertTrue(response.data['waffle_flags'])
-        # This is no longer the first flag
-        #self.assertEqual(response.data['waffle_flags'][0]['name'], 'test.flag')
+        waffle_names = [waffle["name"] for waffle in response.data['waffle_flags']]
+        self.assertIn('test.flag', waffle_names)
 
     @override_switch('test.switch', True)
     def test_response_with_waffle_switch(self):
-        response = self._get_toggle_state_response(is_staff=True)
+        response = self._get_toggle_state_response()
         self.assertIn('waffle_switches', response.data)
         self.assertTrue(response.data['waffle_switches'])
-        # This is no longer the first switch
-        #self.assertEqual(response.data['waffle_switches'][0]['name'], 'test.switch')
+        waffle_names = [waffle["name"] for waffle in response.data['waffle_switches']]
+        self.assertIn('test.switch', waffle_names)
+
+    def test_response_contains_setting_toggles(self):
+        response = self._get_toggle_state_response()
+        self.assertIn("setting_toggles", response.data)
+        self.assertIn("setting_dict_toggles", response.data)
+
+    def test_response_with_setting_toggle(self):
+        _toggle = SettingToggle("MYSETTING", default=False, module_name="module1")
+        with override_settings(MYSETTING=True):
+            response = self._get_toggle_state_response()
+
+        self.assertIn(
+            {
+                "name": "MYSETTING",
+                "is_active": True,
+                "module": "module1",
+                "class": "SettingToggle",
+            },
+            response.data["setting_toggles"],
+        )
+
+    def test_response_with_existing_setting_dict_toggle(self):
+        response = self._get_toggle_state_response()
+        self.assertIn(
+            {
+                "name": "FEATURES",
+                "key": "MILESTONES_APP",
+                "is_active": True,
+                "module": "util.milestones_helpers",
```

Comment:
Should we mock code_owner code somewhere to test getting a result?

Reply:
I don't think it's necessary as under the hood the view makes use of `_add_toggle_instance_details`. It should be enough to expect that this method is correctly unit-tested.

(well, I don't think it is unit-tested, but it's worked until now, right?)

Reply:
The toggle state endpoint was implemented Hackathon style. Ensuring proper coverage should be one of your tasks. 🙂